pkg/scalers/datadog_scaler.go

```diff
@@ -46,12 +46,14 @@ type datadogMetadata struct {
 
 var datadogLog = logf.Log.WithName("datadog_scaler")
 
-var aggregator, filter, rollup *regexp.Regexp
+var aggregator, filter, rollup, ddfuncpresent, ddfunc *regexp.Regexp
 
 func init() {
-	aggregator = regexp.MustCompile(`^(avg|sum|min|max):.*`)
+	aggregator = regexp.MustCompile(`^([\w]+\()?(avg|sum|min|max):.*`)
```

Comment:
Pardon my ignorance, just looking at all these regexes, isn't there any validator in any Datadog library?

Reply:
An excellent question; will check!

Reply:
Had a quick look around, plenty for submitting metrics etc, but not a lot in the way of validating if a metrics query is valid.

The Datadog UI posts to https://app.datadoghq.com/api/v2/query/timeseries when changing queries interactively, and returns 400 when it's invalid - but seeing as this requires auth anyway, if going down that route, you may as well skip validation in the Scaler, and just handle errors when they come back from the upstream.